src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -3177,36 +3177,31 @@ private void ConvertMarkTimes()
             {
                 foreach (KeyValuePair<int, MarkInfo> item in PerHeapMarkTimes)
                 {
-                    if (item.Value.MarkTimes[(int)MarkRootType.MarkSizedRef] == 0.0)
+                    List<(MarkRootType, double)> perRootMarkTime = new List<(MarkRootType, double)>();
+
+                    // Accumulate all the times that have a corresponding mark event.
+                    for (int i = 0; i < item.Value.MarkTimes.Length; i++)
```

Comment:
it seems like it wouldn't take much work to improve the perf here? you could preallocate `perRootMarkTime` instead of allocating one over and over again in per iteration per GC. you could also sort in place instead of creating new tuples every time.

Reply:
I think these are good recommendations.  One of the things that we're seeing more often these days is larger traces from machines with larger heaps.  I think there are a few simple perf/memory improvements here that can make a difference here without you having to write too much code.